subcommand/lifecycle-sidecar/command.go

```diff
@@ -96,14 +93,69 @@ func (c *Command) Run(args []string) int {
 	//
 	// The loop will only exit when the Pod is shut down and we receive a SIGINT.
 	for {
+		// https://github.com/hashicorp/consul-k8s/issues/275
+		// Changing to poll per sync-period and only re-register if service is
+		// no longer registered.
+
+		// 275: get registered services
+		c.consulCommand = []string{"consul", "catalog", "services"}
 		cmd := exec.Command(c.flagConsulBinary, c.consulCommand...)
 
-		// Run the command and record the stdout and stderr output
+		// 275: Poll consul
 		output, err := cmd.CombinedOutput()
 		if err != nil {
-			logger.Error("failed to sync service", "output", strings.TrimSpace(string(output)), "err", err)
+			logger.Error("failed to poll services", "output", strings.TrimSpace(string(output)), "err", err)
 		} else {
-			logger.Info("successfully synced service", "output", strings.TrimSpace(string(output)))
+			logger.Info("successfully polled services", "output", strings.TrimSpace(string(output)))
+		}
+
+		// 275: parse list of registered services from local hcl file.
+		// We set devMode to true so we can get the basic valid default
+		// configuration. devMode doesn't set any services by default so this
+		// is okay since we only look at services.
+		devMode := true
+		b, err := config.NewBuilder(config.BuilderOpts{
```

Comment:
This block is where I'd need to find an alternative approach to getting the service name. @ishustava